javascript/src/index.ts

```diff
@@ -813,6 +811,25 @@ export function getMissingDeps<T>(doc: Doc<T>, heads: Heads): Heads {
     return state.handle.getMissingDeps(heads)
 }
 
+export function splice<T>(doc: Doc<T>, prop: Prop, index: number, del: number, newText?: string) {
+    if (!Reflect.get(doc, IS_PROXY)) {
+      throw new RangeError("object cannot be modified outside of a change block")
+    }
+    const state = _state(doc, false)
+    const objectId = _obj(doc)
+    if (!objectId) {
+      throw new RangeError("invalid object for splice")
+    }
+    const value = state.handle.getWithType(objectId, prop)
+    if (value === null) {
+        throw new RangeError("Cannot splice, not a valid value");
+    } else if (value[0] === 'text') {
+        return state.handle.splice(value[1], index, del, newText)
+    } else {
+        throw new RangeError(`Cannot splice, value is of type '${value[0]}', must be 'text'`);
```

Comment:
I made a test specifically for this - knowing people will try to splice strings in documents that predate this change.   I wanted to make sure the error message was clear instead of the lower level one coming out of the wasm api.